### PR TITLE
[MODORG-37] - Logging improvement

### PR DESCRIPTION
**Pull request by azizbekxm**
## Purpose

https://issues.folio.org/browse/MODORG-37

## Approach

The logging commands of the module has been adjusted according to the document: https://wiki.folio.org/display/DD/Logging 
#### TODOS and Open Questions

 - [ ] Log message must be informative and contain business parameters/indicators (see doc. p. 1.1)
 - [ ] Logging of import business logic actions and steps (see doc. p. 1.2)
 - [ ] Logging of any edge cases scenarios and essential conditional business logic paths (see doc. p. 1.3)
 - [ ] Logging of all exceptional cases (see doc. p. 1.4)
 - [ ] Avoid logging sensitive information or use a specific method of Folio Spring Base to produce messages (see doc. p. 3)

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of inter-module and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by azizbekxm** [comment]
1. Is it okay to also fix grammar issues in PULL_REQUEST_TEMPLATE.md
image

File: src/main/java/org/folio/service/BaseService.java [hunk 1]
```diff
@@ -114,20 +112,21 @@ public CompletableFuture<String> handlePostRequest(JsonObject recordData, String
...
-        logger.debug("Sending 'POST {}' with body: {}", endpoint, recordData.encodePrettily());
+        logger.debug("Trying to create object by endpoint '{}' and body '{}'", endpoint, recordData.encodePrettily());
       }
       httpClient.request(HttpMethod.POST, recordData.toBuffer(), endpoint, okapiHeaders)
         .thenApply(this::verifyAndExtractRecordId)
         .thenAccept(id -> {
           future.complete(id);
-          logger.debug("'POST {}' request successfully processed. Record with '{}' id has been created", endpoint, id);
+          logger.debug("Object was successfully created. Record with '{}' id has been created", id);
         })
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/BaseService.java:122]: which one is more informative 
'POST {}' request su...
Object was.....

File: src/main/java/org/folio/service/BaseService.java [hunk 2]
```diff
@@ -200,19 +200,19 @@ public CompletableFuture<Void> handlePutRequest(String endpoint, JsonObject reco
...
-          logger.error(errorMessage, t);
+          // logger.error("Error deleting '{}' object", endpoint, e);
+          logger.error("Delete '{}' request failed", endpoint, t);
           future.completeExceptionally(t);
           return null;
         });
     } catch (Exception e) {
-      logger.error(String.format(EXCEPTION_CALLING_ENDPOINT_MSG, HttpMethod.DELETE, endpoint), e);
+      logger.debug("Error deleting object by endpoint '{}'", endpoint, e);
       future.completeExceptionally(e);
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/BaseService.java:216]: 1 . Should I use two loggers in both parts
` exceptionally(t -> `
`catch(e) {`  or just one part
2. Which is a more useful loge 
`logger.error("Error deleting '{}' object", endpoint, e);`
`logger.error("Delete '{}' request failed", endpoint, t);`

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java [hunk 3]
```diff
@@ -9,7 +9,6 @@
 import static org.folio.util.ResourcePathResolver.resourcesPath;
 
 import java.util.Collections;
-import java.util.HashSet;
 import java.util.Map;
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/organization/OrganizationStorageService.java:12]: looks it is not used somewhere

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java [hunk 4]
```diff
@@ -44,9 +46,12 @@ public class OrganizationStorageService extends BaseService implements Organizat
   @Override
   public CompletableFuture<Organization> createOrganization(Organization organization, Context context,
       Map<String, String> headers) {
+    // Should I add the organization object to log
+    logger.debug("createOrganization:: Trying to create organization with name: {}", organization.getName());
     HttpClientInterface client = getHttpClient(headers);
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/organization/OrganizationStorageService.java:51]: Should I add the organization object to the log? Organization to the string may not be implemented. I am using only its name

File: src/main/java/org/folio/service/BaseService.java (hunk 1 above)
Comment (SerhiiNosko) [reply src/main/java/org/folio/service/BaseService.java:122]: @azizbekxm  the same structure as you did in mod-gobi, so lets use second one

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java (hunk 3 above)
Comment (SerhiiNosko) [reply src/main/java/org/folio/service/organization/OrganizationStorageService.java:12]: If you see some unused imports - feel free to delete them

File: src/main/java/org/folio/service/BaseService.java (hunk 1 above)
Comment (azizbekxm) [reply src/main/java/org/folio/service/BaseService.java:122]: ok, Thanks

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java (hunk 4 above)
Comment (SerhiiNosko) [reply src/main/java/org/folio/service/organization/OrganizationStorageService.java:51]: Basically organization id will be the most useful in logs, additionally you can provide organization name

File: src/main/java/org/folio/service/BaseService.java (hunk 2 above)
Comment (Dmitriy-Butramyou) [reply src/main/java/org/folio/service/BaseService.java:213]: I think it will be better with 2 loggers.

Comment (SerhiiNosko) [reply src/main/java/org/folio/service/BaseService.java:213]: Yes, I think so, there is possible situation that this exceptionally completed feature can be logged on higher level, but in any case I think we should always log catch causes

File: src/main/java/org/folio/service/BaseService.java [hunk 5]
```diff
@@ -200,19 +198,18 @@ public CompletableFuture<Void> handlePutRequest(String endpoint, JsonObject reco
...
         .exceptionally(t -> {
-          String errorMessage = String.format(EXCEPTION_CALLING_ENDPOINT_MSG, HttpMethod.DELETE, endpoint);
-          logger.error(errorMessage, t);
+          logger.error("Object cannot be deleted with using endpoint: {}", endpoint, t);
           future.completeExceptionally(t);
           return null;
         });
     } catch (Exception e) {
-      logger.error(String.format(EXCEPTION_CALLING_ENDPOINT_MSG, HttpMethod.DELETE, endpoint), e);
+      logger.debug("Error deleting object by endpoint '{}'", endpoint, e);
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/BaseService.java:212]: Why it's debug level?

Comment (azizbekxm) [reply src/main/java/org/folio/service/BaseService.java]: sorry, it is a mistake. changed to error

**Review by Dmitriy-Butramyou — APPROVED** [review]

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java [hunk 6]
```diff
@@ -44,9 +46,11 @@ public class OrganizationStorageService extends BaseService implements Organizat
   @Override
   public CompletableFuture<Organization> createOrganization(Organization organization, Context context,
       Map<String, String> headers) {
+    logger.debug("createOrganization:: Trying to create organization with name: {}", organization.getName());
     HttpClientInterface client = getHttpClient(headers);
     CompletableFuture<Organization> future = new CompletableFuture<>();
     if (isSameAccountNumbers(organization)) {
+      logger.warn("crateOrganization:: Account number is not unique");
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/organization/OrganizationStorageService.java:53]: can we add account number from organization for this log?

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java [hunk 7]
```diff
@@ -123,15 +133,18 @@ public CompletableFuture<OrganizationCollection> getOrganizationCollection(int o
   @Override
   public CompletableFuture<Void> updateOrganizationById(String id, Organization updatedOrganization, String lang, Context context,
       Map<String, String> headers) {
+    logger.debug("updateOrganization:: Trying to update organization with id: {}", id);
     CompletableFuture<Void> future = new CompletableFuture<>();
     if (isEmpty(updatedOrganization.getId())) {
       updatedOrganization.setId(id);
     } else if (!id.equals(updatedOrganization.getId())) {
+      logger.warn("updateOrganization:: Mismatch between id in path and request body");
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/organization/OrganizationStorageService.java:141]: Please add these 2 ids to log

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java [hunk 8]
```diff
@@ -123,15 +133,18 @@ public CompletableFuture<OrganizationCollection> getOrganizationCollection(int o
...
       future.completeExceptionally(new HttpException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY.toInt(),
         MISMATCH_BETWEEN_ID_IN_PATH_AND_BODY.toError()));
       return future;
     }
     if (isSameAccountNumbers(updatedOrganization)) {
+      logger.warn("updateOrganization:: Account number is not unique");
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/organization/OrganizationStorageService.java:147]: please add accountNumber to this log

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java (hunk 6 above)
Comment (azizbekxm) [reply src/main/java/org/folio/service/organization/OrganizationStorageService.java:53]: I guess List<Account> accounts don't have to String method in the Account class
`isSameAccountNumbers(organization)` check account numbers from the organization based on the unique size and original size.
I can add the account numbers size

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java (hunk 7 above)
Comment (azizbekxm) [reply src/main/java/org/folio/service/organization/OrganizationStorageService.java:141]: ok, I fixed

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java [hunk 9]
```diff
@@ -159,10 +174,12 @@ public CompletableFuture<Void> updateOrganizationById(String id, Organization up
...
   public CompletableFuture<Void> deleteOrganizationById(String id, Context context, Map<String, String> headers) {
+    logger.debug("deleteOrganizationById:: Trying to delete organization by id: {}", id);
     HttpClientInterface client = getHttpClient(headers);
     return handleDeleteRequest(resourceByIdPath(ORGANIZATIONS, id), client, headers, logger).handle((org, t) -> {
       client.closeClient();
       if (Objects.nonNull(t)) {
+        logger.warn("Error deleting organization with id: {}", id, t);
         throw new CompletionException(t.getCause());
       }
       return null;
```
Comment (azizbekxm) [inline src/main/java/org/folio/service/organization/OrganizationStorageService.java:185]: is it okay to use logger.warn only inside of object.nonNull?
Or Should I use the above return null?

File: src/main/java/org/folio/service/BaseService.java [hunk 10]
```diff
@@ -173,20 +170,21 @@ public CompletableFuture<Void> handlePutRequest(String endpoint, JsonObject reco
...
     try {
       if (logger.isDebugEnabled()) {
-        logger.debug("Sending 'PUT {}' with body: {}", endpoint, recordData.encodePrettily());
+        logger.debug("Trying to update object by endpoint '{}' and body '{}'", endpoint, recordData.encodePrettily());
       }
       httpClient.request(HttpMethod.PUT, recordData.toBuffer(), endpoint, okapiHeaders)
         .thenApply(this::verifyAndExtractBody)
         .thenAccept(response -> {
-          logger.debug("'PUT {}' request successfully processed", endpoint);
+          logger.info("Object was successfully updated. Record with '{}' id has been updated", endpoint);
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/BaseService.java:178]: please change to debug level here

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java [hunk 11]
```diff
@@ -67,13 +72,14 @@ private boolean isSameAccountNumbers(Organization organization) {
     Set<String> uniqueAccounts = organization.getAccounts().stream()
       .map(Account::getAccountNo)
       .collect(Collectors.toSet());
-
+    logger.debug("isSameAccountNumbers:: match unique accounts numbers size '{}' with accounts size '{}'", uniqueAccounts, organization.getAccounts().size());
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/organization/OrganizationStorageService.java:75]: need to change uniqueAccounts -> uniqueAccounts .size()

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java (hunk 8 above)
Comment (SerhiiNosko) [reply src/main/java/org/folio/service/organization/OrganizationStorageService.java:147]: if its not possbile to add acount number - add orginazation id here

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java (hunk 11 above)
Comment (azizbekxm) [reply src/main/java/org/folio/service/organization/OrganizationStorageService.java:75]: oh, thanks fixed

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java (hunk 9 above)
Comment (SerhiiNosko) [reply src/main/java/org/folio/service/organization/OrganizationStorageService.java:185]: Its okay to use only  inside of object.nonNull

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java (hunk 6 above)
Comment (azizbekxm) [reply src/main/java/org/folio/service/organization/OrganizationStorageService.java:53]: ok I am adding organization name

File: src/main/java/org/folio/service/BaseService.java (hunk 10 above)
Comment (azizbekxm) [reply src/main/java/org/folio/service/BaseService.java:178]: ok, fixed

File: src/main/java/org/folio/service/BaseService.java [hunk 12]
```diff
@@ -138,25 +135,25 @@ public CompletableFuture<JsonObject> handleGetRequest(String endpoint, HttpClien
 
     CompletableFuture<JsonObject> future = new CompletableFuture<>();
     try {
-      logger.info("Calling GET {}", endpoint);
+      logger.debug("Trying to get '{}' object", endpoint);
       httpClient.request(HttpMethod.GET, endpoint, okapiHeaders)
         .thenApply(response -> {
-          logger.debug("Validating response for GET {}", endpoint);
+          logger.info("Validating response for get request '{}'", endpoint);
```
[the remark this answers is not in the document]
Comment (azizbekxm) [reply src/main/java/org/folio/service/BaseService.java:141]: ok, thanks fixed

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java (hunk 9 above)
Comment (azizbekxm) [reply src/main/java/org/folio/service/organization/OrganizationStorageService.java:185]: thanks

File: src/main/java/org/folio/service/organization/OrganizationStorageService.java (hunk 8 above)
Comment (azizbekxm) [reply src/main/java/org/folio/service/organization/OrganizationStorageService.java:147]: yeah, thanks added

**Review by SerhiiNosko — APPROVED** [review]

**Pull request merged**